### PR TITLE
fix checkpoint generated by ddp training cannot be loaded

### PR DESCRIPTION
**Pull request by yzslab**
Hi,
This pull request can fix https://github.com/nerfstudio-project/nerfstudio/issues/1636, and will also allow the checkpoint from non-ddp training to be loaded in ddp mode.

### CODE REVIEW

File: nerfstudio/pipelines/base_pipeline.py [hunk 1]
```diff
@@ -108,9 +108,14 @@ def device(self):
         return self.model.device
 
     def load_state_dict(self, state_dict: Mapping[str, Any], strict: bool = True):
-        model_state = {key[len("_model.") :]: value for key, value in state_dict.items() if key.startswith("_model.")}
+        # remove the "_model." or "_model.module." prefix from key
+        model_state = {
+            key[len("_model.module.") :] if key.startswith("_model.module.") else key[len("_model.") :]: value
+            for key, value in state_dict.items()
+            if key.startswith("_model.")
+        }
```
Comment (tancik) [inline nerfstudio/pipelines/base_pipeline.py:116]: This comprehension is getting kinda messy and hard to read. Can you rewrite as a for loop

**Review by tancik — COMMENTED** [review]
Have you tested that the existing loading still works?

**Comment by yzslab** [comment]
> Have you tested that the existing loading still works?

Hi, this PR is compatible with the checkpoints generated by previous version. I haved tested loading the non-DDP and DDP checkpoint generated by previous version in both non-DDP and DDP mode of this PR, it works fine.

**Review by tancik — APPROVED** [review]
LGTM. Thanks for fixing this bug and adding a test!

**Pull request merged**